experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,199 @@
+    output_run_path.append(str(random_seed))
+    #get the output run path
+    output_run_path = os.path.abspath(os.path.join('/data', 'Pipeline_Run', 
+                                                   '_'.join(output_run_path)+'.yaml'))
+    #create the directory
+    os.makedirs(os.path.dirname(output_run_path),exist_ok=True)
+    #evaluate pipeline
+    evaluate_pipeline_via_d3m_cli(pipeline=pipeline, problem=problem,
+        input=input, output_run=output_run_path,
+        random_seed=random_seed, data_pipeline_path=data_pipeline_path,
+        data_random_seed=data_random_seed, data_params=data_params, 
+        scoring_pipeline=scoring_pipeline, scoring_params=scoring_params,
+        scoring_random_seed=scoring_random_seed)
+
+def evaluate_pipeline_via_d3m_cli(pipeline: str=None,
+    problem: str=None,
+    input: str=None,
+    output_run: str=None,
+    random_seed: int=0,
+    data_pipeline_path: str=k_fold_split_path,
+    data_random_seed: int=0,
+    data_params=None,
+    scoring_pipeline: str=None,
+    scoring_params=None,
+    scoring_random_seed: int=0):
+    """ 
+    Evaluate pipeline on problem using d3m's runtime cli. 
+    Wrapper function to execute d3m's runtime cli 'evaluate' command.
+    Arguments mirror the same arguments using the cli.
+    Only handles cases with a data preparation pipeline in the 
+    pipeline run.
+
+    Parameters
+    ----------
+    pipeline : path_like or uuid4 str
+        path to pipeline doc or pipeline ID
+    problem : path_like str
+        path to problem doc
+    input : path_like str
+        path to input full data
+    output_run : path_like str or '-'
+        path where pipeline_run doc
+        will be saved.
+        use '-' for stdin
+    random_seed : int
+        random seed to used for
+        pipeline run
+    data_pipeline_path: str
+        path to data prepation pipeline
+    data_random_seed: int
+        random_seed to be used in data preparation
+    data_params:
+        parameters for data preparation
+    scoring_params:
+        parameters for scoring pipeline
+    scoring_random_seed: int
+        random seed for scoring
+    scoring_pipeline: str
+        path to scoring pipeline
+    Return:
+    -------
+    None
+    
+    Raises:
+    -------
+    ValueError
+        when parameter value is
+        invalid
+    """    
+    if (not os.path.isfile(pipeline)):
+        raise ValueError('\'{}\' param not a file path'.format('pipeline'))
+
+    if (not os.path.isfile(problem)): 
+        raise ValueError('\'{}\' param not a file path'.format('problem'))
+
+    if (not os.path.isfile(input)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
+        
+    if (not os.path.isfile(data_pipeline_path)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
+        
+    if (not os.path.isfile(scoring_pipeline)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
```

Comment:
Let's use custom exceptions everywhere, from `exceptions.py`. Try to use the existing exceptions, but you can make a new exception class there if needed.